pages/k8s/cni-calico.md

```diff
@@ -0,0 +1,368 @@
+choose 0.0.0.0 for rack 0 and 0.0.0.1 for rack 1.
+
+Configure the route reflectors:
+
+```bash
+juju config calico route-reflector-cluster-ids="
+0: 0.0.0.0
+1: 0.0.0.0
+2: 0.0.0.1
+3: 0.0.0.1
+"
+```
+
+Configure route reflectors to peer with the top of rack switches:
+
+```bash
+juju config calico unit-bgp-peers="
+0:
+- address: 10.0.0.1
+  as-number: 64512
+1:
+- address: 10.0.1.1
+  as-number: 64512
+2:
+- address: 10.0.2.1
+  as-number: 64513
+3:
+- address: 10.0.3.1
+  as-number: 64513
+"
+```
+
+Configure Calico nodes to peer with the route reflectors:
+
+```bash
+juju config calico subnet-bgp-peers="
+10.0.0.0/24:
+- address: 10.0.0.2
+  as-number: 64512
+- address: 10.0.1.2
+  as-number: 64512
+10.0.1.0/24:
+- address: 10.0.0.2
+  as-number: 64512
+- address: 10.0.1.2
+  as-number: 64512
+10.0.2.0/24:
+- address: 10.0.2.2
+  as-number: 64513
+- address: 10.0.3.2
+  as-number: 64513
+10.0.3.0/24:
+- address: 10.0.2.2
+  as-number: 64513
+- address: 10.0.3.2
+  as-number: 64513
+"
+```
+
+And disable the node-to-node mesh:
+
+```bash
+juju config calico node-to-node-mesh=false
+```
+
+You will also need to configure the top of rack switches to peer with the
+Calico route reflectors. Instructions for doing this should be provided in the
+documentation for the specific model of switch you are using.
+
+## Configuring multiple Calico pools
+
+The Calico charm creates a single IPPool. If multiple IPPools pools are desired,
+then you must first disable the charm's built-in pool management:
+
+```bash
+juju config calico manage-pools=false
+```
+
+Then use calicoctl to create your own pools. For example:
+
+```bash
+juju ssh calico/0 calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: IPPool
+metadata:
+  name: my-pool
+spec:
+  cidr: 192.168.0.0/24
+  ipipMode: Never
+  natOutgoing: true
+EOF
+```
+
+## Using a private Docker registry
+
+For a general introduction to using a private Docker registry with **CDK**, please
+refer to the [Private Docker Registry][] page.
+
+In addition to the steps documented there, you will need to upload the
+following images to the registry:
+
+```no-highlight
+docker.io/calico/node:v3.6.1
+docker.io/calico/kube-controllers:v3.6.1
+```
+
+And configure Calico to use the registry:
+
+```bash
+export IP=`juju run --unit docker-registry/0 'network-get website --ingress-address'`
+export PORT=`juju config docker-registry registry-port`
+export REGISTRY=$IP:$PORT
+juju config calico \
+  calico-node-image=$registry/calico/node:v3.6.1 \
+  calico-policy-image=$registry/calico/kube-controllers:v3.6.1
+```
+
+## Troubleshooting
+
+If there is an issue with connectivity, it can be useful to inspect the Juju logs.
+To see a complete set of logs for Calico
+
+```bash
+juju debug-log --replay --include=calico
+```
+
+For additional troubleshooting pointers, please see the [dedicated troubleshooting page][troubleshooting].
+
+
+## Useful links
+
+-
```

Comment:
"Useful links" section is empty. Only potentially useful link I can think of is the Calico docs: https://docs.projectcalico.org/v3.6/introduction/